homeassistant/components/drop_connect/select.py

```diff
@@ -0,0 +1,96 @@
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONF_DEVICE_TYPE, DEV_HUB, DOMAIN
+from .coordinator import DROPDeviceDataUpdateCoordinator
+from .entity import DROPEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+# Select type constants
+PROTECT_MODE = "protect_mode"
+
+PROTECT_MODE_OPTIONS = ["AWAY", "HOME", "SCHEDULE"]
+
+FLOOD_ICON = "mdi:home-flood"
+
+
+@dataclass(kw_only=True, frozen=True)
+class DROPSelectEntityDescription(SelectEntityDescription):
+    """Describes DROP select entity."""
+
+    value_fn: Callable[[DROPDeviceDataUpdateCoordinator], str | None]
+    set_fn: Callable[[DROPDeviceDataUpdateCoordinator, str], Any]
+
+
+SELECTS: list[DROPSelectEntityDescription] = [
+    DROPSelectEntityDescription(
+        key=PROTECT_MODE,
+        translation_key=PROTECT_MODE,
+        icon=FLOOD_ICON,
+        options=PROTECT_MODE_OPTIONS,
+        value_fn=lambda device: device.drop_api.protect_mode(),
+        set_fn=lambda device, value: device.set_protect_mode(value),
+    )
+]
+
+# Defines which selects are used by each device type
+DEVICE_SELECTS: dict[str, list[str]] = {
+    DEV_HUB: [PROTECT_MODE],
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the DROP selects from config entry."""
+    _LOGGER.debug(
+        "Set up select for device type %s with entry_id is %s",
+        config_entry.data[CONF_DEVICE_TYPE],
+        config_entry.entry_id,
+    )
+
+    if config_entry.data[CONF_DEVICE_TYPE] in DEVICE_SELECTS:
+        async_add_entities(
+            DROPSelect(hass.data[DOMAIN][config_entry.entry_id], select)
+            for select in SELECTS
+            if select.key in DEVICE_SELECTS[config_entry.data[CONF_DEVICE_TYPE]]
+        )
+
+
+class DROPSelect(DROPEntity, SelectEntity):
+    """Representation of a DROP select."""
+
+    entity_description: DROPSelectEntityDescription
+
+    def __init__(
+        self,
+        coordinator: DROPDeviceDataUpdateCoordinator,
+        entity_description: DROPSelectEntityDescription,
+    ) -> None:
+        """Initialize the select."""
+        super().__init__(entity_description.key, coordinator)
+        self.entity_description = entity_description
+
+    @property
+    def current_option(self) -> str | None:
+        """Return the current selected option."""
+        return self.entity_description.value_fn(self.coordinator)
+
+    async def async_select_option(self, option: str) -> None:
+        """Update the current selected option."""
+        if option in PROTECT_MODE_OPTIONS:
+            await self.entity_description.set_fn(self.coordinator, option)
```

Comment:
Please raise `ServiceValidationError` is the option is not in `PROTECT_MODE_OPTIONS`
Note that a translation key can be added to support translations.

Reply:
Thanks for the translation key suggestion.  I added that in 7a031bcff0c9631def3e3968b9fe045a5d322dc9.  I'm glad I did that now because I had to push the values to lower-case to appease one of the linters, and that meant changing the API in my device firmware.  Much easier before it is released.

If I add an `else: raise ServiceValidationError` to `async_select_option()`, I can't seem to hit that line in my tests.  If I call `hass.services.async_call()` with an invalid value in a test, something else raises a `ValueError` exception before `async_select_option()` even fires.  It seems like I should just remove the `if option in PROTECT_MODE_OPTIONS:` test altogether because it seems to be always true.  Thoughts?

Reply:
My fault. It seems SelectEntity does a validation check that calls value Error.
This means you can remove the check to see if the option is valid here, and there is no need to raise.
```suggestion
        await self.entity_description.set_fn(self.coordinator, option)
```

Reply:
Thanks for verifying.  I removed the check in 0a100876bb02c292b07cfc9135615e5f471255f7.